### PR TITLE
refactor(ndt_scan_matcher): modularize pose initialization

### PR DESCRIPTION
**Pull request by kminoda**
## Description
Aiming at integrating dynamic map loading feature in ndt_scan_matcher smoothly, I would like to modularize ndt_scan_matcher into the following modules:

- module that load pointcloud map (map_module) (Done)
- module that execute the pose initialization (initial_pose_module) (This PR)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: localization/ndt_scan_matcher/src/pose_initialization_module.cpp [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+  auto best_particle_ptr = std::max_element(
+    std::begin(particle_array), std::end(particle_array),
+    [](const Particle & lhs, const Particle & rhs) { return lhs.score < rhs.score; });
+
+  geometry_msgs::msg::PoseWithCovarianceStamped result_pose_with_cov_msg;
+  result_pose_with_cov_msg.header.stamp = initial_pose_with_cov.header.stamp;
+  result_pose_with_cov_msg.header.frame_id = map_frame_;
+  result_pose_with_cov_msg.pose.pose = best_particle_ptr->result_pose;
+  // ndt_pose_with_covariance_pub_->publish(result_pose_with_cov_msg);
```
Comment (KYabuuchi) [inline localization/ndt_scan_matcher/src/pose_initialization_module.cpp:126]: Please remove this comment line. 
`is_local_optimal_solution_oscillation` is not defined.

Comment (kminoda) [reply localization/ndt_scan_matcher/src/pose_initialization_module.cpp]: Thanks! Done: https://github.com/autowarefoundation/autoware.universe/pull/2254/commits/2b8359f8054805e39f3e5c9dc63f3fb091fb813f

(Is the comment `is_local_optimal_solution_oscillation is not defined.` is just a mistake?)

Comment (KYabuuchi) [reply localization/ndt_scan_matcher/src/pose_initialization_module.cpp]: Nice!
(Yes. Sorry. My comment was mistake. :cry: )

**Review by KYabuuchi — APPROVED** [review]
LGTM

**Pull request merged**